File: backend/internal/model/mortalidade.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// Mortalidade representa um registro de mortalidade de aves em um lote.
type Mortalidade struct {
	ID         uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	LoteID     uuid.UUID `gorm:"type:uuid;not null;index:idx_mortalidades_lote_data" json:"lote_id" validate:"required"`
	Data       time.Time `gorm:"type:date;not null;index:idx_mortalidades_lote_data" json:"data" validate:"required"`
	Quantidade int       `gorm:"type:integer;not null" json:"quantidade" validate:"required,gt=0"`
	Causa      string    `gorm:"type:varchar(50);default:'Indefinida'" json:"causa" validate:"omitempty,oneof=Indefinida Ascite 'Morte Subita' 'Problemas Locomotores' Refugo Onfalite Infecciosa Outra"`
	Observacao *string   `gorm:"type:text" json:"observacao,omitempty"`
	FotoURL    *string   `gorm:"type:text;column:foto_url" json:"foto_url,omitempty"`
	CreatedAt  time.Time `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt  time.Time `gorm:"not null;default:now()" json:"updated_at"`

	// Relacionamentos
	Lote Lote `gorm:"foreignKey:LoteID" json:"-"`
}

func (Mortalidade) TableName() string {
	return "mortalidades"
}

func (m *Mortalidade) BeforeCreate(tx *gorm.DB) error {
	if m.ID == uuid.Nil {
		m.ID = uuid.New()
	}
	if m.Causa == "" {
		m.Causa = "Indefinida"
	}
	return nil
}

File: flutter_app/lib/features/lote_detail/domain/entities/mortalidade.dart
/// Registro de mortalidade diaria de um lote.
///
/// Campos mapeados a partir do endpoint GET /lotes/{id}/mortalidades.
/// Backend retorna: id, lote_id, data, quantidade, causa, observacao?, created_at.
class Mortalidade {
  final String id;
  final String loteId;
  final String data;

  /// Quantidade de aves mortas.
  final int quantidade;

  /// Causa da mortalidade.
  final String causa;

  /// Observacao adicional (opcional).
  final String? observacao;

  final String createdAt;

  const Mortalidade({
    required this.id,
    required this.loteId,
    required this.data,
    required this.quantidade,
    required this.causa,
    this.observacao,
    required this.createdAt,
  });

  factory Mortalidade.fromJson(Map<String, dynamic> json) {
    return Mortalidade(
      id: json['id'] as String,
      loteId: json['lote_id'] as String,
      data: json['data'] as String,
      quantidade: (json['quantidade'] as num).toInt(),
      causa: json['causa'] as String? ?? 'Indefinida',
      observacao: json['observacao'] as String?,
      createdAt: json['created_at'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'lote_id': loteId,
      'data': data,
      'quantidade': quantidade,
      'causa': causa,
      if (observacao != null) 'observacao': observacao,
      'created_at': createdAt,
    };
  }
}
